import sys
from PyQt5.QtGui import *
from PyQt5.QtCore import *
from PyQt5.QtGui import *
from PyQt5 import QtWidgets,QtCore,QtGui
import regex as re
import sheet_compiler as sscomp
import sheet_compiler2 as sscomp2

class Ui_MainWindow(object):
    def __init__(self):
        super().__init__()

    def SSBrowseClicked(self):
        pass

    def SSAcceptClicked(self):
        pass

    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow")
        MainWindow.resize(1158, 681)
        self.centralwidget = QtWidgets.QWidget(MainWindow)
        self.centralwidget.setObjectName("centralwidget")
        self.tabWidget = QtWidgets.QTabWidget(self.centralwidget)
        self.tabWidget.setGeometry(QtCore.QRect(0, 0, 2331, 1081))
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Preferred, QtWidgets.QSizePolicy.Preferred)
        sizePolicy.setHorizontalStretch(True)
        sizePolicy.setVerticalStretch(True)
        sizePolicy.setHeightForWidth(self.tabWidget.sizePolicy().hasHeightForWidth())
        self.tabWidget.setSizePolicy(sizePolicy)
        self.tabWidget.setObjectName("tabWidget")

############# ANALISI TAB ######################################        
        self.Analisitab = QtWidgets.QWidget()
        self.Analisitab.setAccessibleName("")
        self.Analisitab.setObjectName("Analisitab")
        self.groupBox_2 = QtWidgets.QGroupBox(self.Analisitab)
        self.groupBox_2.setGeometry(QtCore.QRect(70, 210, 401, 311))
        self.groupBox_2.setAlignment(QtCore.Qt.AlignCenter)
        self.groupBox_2.setFlat(False)
        self.groupBox_2.setCheckable(False)
        self.groupBox_2.setObjectName("groupBox_2")
        self.formLayout_2 = QtWidgets.QFormLayout(self.groupBox_2)
        self.formLayout_2.setObjectName("formLayout_2")
        self.alignment_check = QtWidgets.QCheckBox(self.groupBox_2)
        self.alignment_check.setChecked(True)
        self.alignment_check.setObjectName("alignment_check")
        self.formLayout_2.setWidget(0, QtWidgets.QFormLayout.LabelRole, self.alignment_check)
        self.read_group_check = QtWidgets.QCheckBox(self.groupBox_2)
        self.read_group_check.setChecked(True)
        self.read_group_check.setObjectName("read_group_check")
        self.formLayout_2.setWidget(2, QtWidgets.QFormLayout.LabelRole, self.read_group_check)
        self.markDup_check = QtWidgets.QCheckBox(self.groupBox_2)
        self.markDup_check.setChecked(True)
        self.markDup_check.setObjectName("markDup_check")
        self.formLayout_2.setWidget(3, QtWidgets.QFormLayout.LabelRole, self.markDup_check)
        self.indel_real_check = QtWidgets.QCheckBox(self.groupBox_2)
        self.indel_real_check.setChecked(True)
        self.indel_real_check.setObjectName("indel_real_check")
        self.formLayout_2.setWidget(4, QtWidgets.QFormLayout.LabelRole, self.indel_real_check)
        self.BQSR_check = QtWidgets.QCheckBox(self.groupBox_2)
        self.BQSR_check.setChecked(True)
        self.BQSR_check.setObjectName("BQSR_check")
        self.formLayout_2.setWidget(5, QtWidgets.QFormLayout.LabelRole, self.BQSR_check)
        spacerItem = QtWidgets.QSpacerItem(20, 60, QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Expanding)
        self.formLayout_2.setItem(6, QtWidgets.QFormLayout.LabelRole, spacerItem)
        self.variant_call_check = QtWidgets.QCheckBox(self.groupBox_2)
        self.variant_call_check.setChecked(True)
        self.variant_call_check.setObjectName("variant_call_check")
        self.formLayout_2.setWidget(7, QtWidgets.QFormLayout.LabelRole, self.variant_call_check)
        self.variant_filter_check = QtWidgets.QCheckBox(self.groupBox_2)
        self.variant_filter_check.setChecked(True)
        self.variant_filter_check.setObjectName("variant_filter_check")
        self.formLayout_2.setWidget(8, QtWidgets.QFormLayout.LabelRole, self.variant_filter_check)
        self.variant_ann_check = QtWidgets.QCheckBox(self.groupBox_2)
        self.variant_ann_check.setChecked(True)
        self.variant_ann_check.setObjectName("variant_ann_check")
        self.formLayout_2.setWidget(9, QtWidgets.QFormLayout.LabelRole, self.variant_ann_check)
        self.CNV_check = QtWidgets.QCheckBox(self.groupBox_2)
        self.CNV_check.setChecked(True)
        self.CNV_check.setObjectName("CNV_check")
        self.formLayout_2.setWidget(10, QtWidgets.QFormLayout.LabelRole, self.CNV_check)
        spacerItem1 = QtWidgets.QSpacerItem(20, 40, QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Expanding)
        self.formLayout_2.setItem(1, QtWidgets.QFormLayout.LabelRole, spacerItem1)
        self.groupBox = QtWidgets.QGroupBox(self.Analisitab)
        self.groupBox.setGeometry(QtCore.QRect(70, 30, 401, 149))
        self.groupBox.setAlignment(QtCore.Qt.AlignCenter)
        self.groupBox.setObjectName("groupBox")
        self.formLayout = QtWidgets.QFormLayout(self.groupBox)
        self.formLayout.setObjectName("formLayout")
        self.label = QtWidgets.QLabel(self.groupBox)
        self.label.setObjectName("label")
        self.formLayout.setWidget(0, QtWidgets.QFormLayout.LabelRole, self.label)
        self.lineEdit = QtWidgets.QLineEdit(self.groupBox)
        self.lineEdit.setObjectName("lineEdit")
        self.formLayout.setWidget(0, QtWidgets.QFormLayout.FieldRole, self.lineEdit)
        self.label_2 = QtWidgets.QLabel(self.groupBox)
        self.label_2.setObjectName("label_2")
        self.formLayout.setWidget(2, QtWidgets.QFormLayout.LabelRole, self.label_2)
        self.comboBox = QtWidgets.QComboBox(self.groupBox)
        self.comboBox.setObjectName("comboBox")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.formLayout.setWidget(2, QtWidgets.QFormLayout.FieldRole, self.comboBox)
        self.comboBox_2 = QtWidgets.QComboBox(self.groupBox)
        self.comboBox_2.setObjectName("comboBox_2")
        self.comboBox_2.addItem("")
        self.comboBox_2.addItem("")
        self.comboBox_2.addItem("")
        self.formLayout.setWidget(4, QtWidgets.QFormLayout.FieldRole, self.comboBox_2)
        self.label_3 = QtWidgets.QLabel(self.groupBox)
        self.label_3.setObjectName("label_3")
        self.formLayout.setWidget(4, QtWidgets.QFormLayout.LabelRole, self.label_3)
        
        self.samplesheet_editor = sscomp2.Samplesheet_editor(self.Analisitab)
        #self.Analisitab.addWidget(self.samplesheet_editor)
        # self.groupBox_3 = QtWidgets.QGroupBox(self.Analisitab)
        # self.groupBox_3.setGeometry(QtCore.QRect(499, 30, 601, 491))
        # self.groupBox_3.setAlignment(QtCore.Qt.AlignCenter)
        # self.groupBox_3.setObjectName("groupBox_3")
        # self.verticalLayoutWidget = QtWidgets.QWidget(self.groupBox_3)
        # self.verticalLayoutWidget.setGeometry(QtCore.QRect(0, 20, 601, 471))
        # self.verticalLayoutWidget.setObjectName("verticalLayoutWidget")
        # self.verticalLayout = QtWidgets.QVBoxLayout(self.verticalLayoutWidget)
        # self.verticalLayout.setContentsMargins(0, 0, 0, 0)
        # self.verticalLayout.setObjectName("verticalLayout")
        # self.frame = QtWidgets.QFrame(self.verticalLayoutWidget)
        # self.frame.setFrameShape(QtWidgets.QFrame.StyledPanel)
        # self.frame.setFrameShadow(QtWidgets.QFrame.Raised)
        # self.frame.setObjectName("frame")
        # self.verticalLayout_2 = QtWidgets.QVBoxLayout(self.frame)
        # self.verticalLayout_2.setObjectName("verticalLayout_2")
        # self.horizontalLayout = QtWidgets.QHBoxLayout()
        # self.horizontalLayout.setObjectName("horizontalLayout")
        # self.label_4 = QtWidgets.QLabel(self.frame)
        # self.label_4.setObjectName("label_4")
        # self.horizontalLayout.addWidget(self.label_4)
        # self.lineEdit_2 = QtWidgets.QLineEdit(self.frame)
        # self.lineEdit_2.setObjectName("lineEdit_2")
        # self.horizontalLayout.addWidget(self.lineEdit_2)

        # self.SSBrowse_Button = QtWidgets.QPushButton(self.frame)
        # self.SSBrowse_Button.setObjectName("SSBrowse_Button")
        # self.horizontalLayout.addWidget(self.SSBrowse_Button)
        # self.SSBrowse_Button.clicked.connect(self.getfile)
        # self.verticalLayout_2.addLayout(self.horizontalLayout)

        # self.table = QtWidgets.QTableWidget(self.frame)
        # self.table.setObjectName("table")
        # self.table.setColumnCount(0)
        # self.table.setRowCount(0)

        # self.verticalLayout_2.addWidget(self.table)
        # self.horizontalLayout_2 = QtWidgets.QHBoxLayout()
        # self.horizontalLayout_2.setObjectName("horizontalLayout_2")
        # spacerItem2 = QtWidgets.QSpacerItem(40, 20, QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Minimum)
        # self.horizontalLayout_2.addItem(spacerItem2)
        # self.SSSave_Button = QtWidgets.QPushButton(self.frame)
        # self.SSSave_Button.setObjectName("SSSave_Button")
        # self.horizontalLayout_2.addWidget(self.SSSave_Button)
        # self.SSSave_Button.clicked.connect(sscomp.savefile)

        # self.verticalLayout_2.addLayout(self.horizontalLayout_2)
        #self.verticalLayout.addWidget(self.frame)
        self.pushButton = QtWidgets.QPushButton(self.Analisitab)
        self.pushButton.setGeometry(QtCore.QRect(70, 550, 1031, 71))
        self.pushButton.setObjectName("pushButton")

        self.tabWidget.addTab(self.Analisitab, "")


############# PANEL TAB ###################################### 
        self.panelstab = QtWidgets.QWidget()
        self.panelstab.setObjectName("panelstab")
        self.tabWidget.addTab(self.panelstab, "")

############# PIPELINE TAB #################################### 
        self.pipelinestab = QtWidgets.QWidget()
        self.pipelinestab.setObjectName("pipelinestab")
        self.tabWidget.addTab(self.pipelinestab, "")

############# TOOLS TAB ####################################
        self.toolstab = QtWidgets.QWidget()
        self.toolstab.setObjectName("toolstab")
        self.tabWidget.addTab(self.toolstab, "")

############# MENU BAR ####################################
        MainWindow.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(MainWindow)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 1158, 22))
        self.menubar.setObjectName("menubar")

        self.menuPowerCall_v2_0 = QtWidgets.QMenu(self.menubar)
        self.menuPowerCall_v2_0.setObjectName("menuPowerCall_v2_0")
        MainWindow.setMenuBar(self.menubar)

        self.retranslateUi(MainWindow)
        self.tabWidget.setCurrentIndex(0)
        #self.SSBrowse_Button.clicked.connect(self.SSBrowseClicked)
        #self.SSAccept_Button.clicked.connect(self.SSAcceptClicked)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)


    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "PowerCall v2.0"))
        self.groupBox_2.setTitle(_translate("MainWindow", "Workflow"))
        self.alignment_check.setText(_translate("MainWindow", "FASTQ Alignment"))
        self.read_group_check.setText(_translate("MainWindow", "Add or Replace Read Groups"))
        self.markDup_check.setText(_translate("MainWindow", "Marking Duplicates"))
        self.indel_real_check.setText(_translate("MainWindow", "Indel Realignement"))
        self.BQSR_check.setText(_translate("MainWindow", "Base Quality Score Recalibration"))
        self.variant_call_check.setText(_translate("MainWindow", "Variant Calling"))
        self.variant_filter_check.setText(_translate("MainWindow", "Variant Filtering"))
        self.variant_ann_check.setText(_translate("MainWindow", "Variant Annotation"))
        self.CNV_check.setText(_translate("MainWindow", "CNV Calling"))
        self.groupBox.setTitle(_translate("MainWindow", "Analysis Informations"))
        self.label.setText(_translate("MainWindow", "Run ID"))
        self.label_2.setText(_translate("MainWindow", "Panel Name"))
        self.comboBox.setItemText(0, _translate("MainWindow", "Illumina TrusighCardio"))
        self.comboBox.setItemText(1, _translate("MainWindow", "Illumina TrusighCancer"))
        self.comboBox.setItemText(2, _translate("MainWindow", "Illumina TrusighOne"))
        self.comboBox.setItemText(3, _translate("MainWindow", "SophiaGen HCS"))
        self.comboBox.setItemText(4, _translate("MainWindow", "Agilent CardioV1"))
        self.comboBox.setItemText(5, _translate("MainWindow", "Agilent AneurysmV1"))
        self.comboBox.setItemText(6, _translate("MainWindow", "Agilent CancerV1"))
        self.comboBox_2.setItemText(0, _translate("MainWindow", "Germline"))
        self.comboBox_2.setItemText(1, _translate("MainWindow", "Somatic"))
        self.comboBox_2.setItemText(2, _translate("MainWindow", "Cellfree"))
        self.label_3.setText(_translate("MainWindow", "Pipeline"))
        self.samplesheet_editor.groupBox_3.setTitle(_translate("MainWindow", "Edit Sample sheet "))
        self.samplesheet_editor.label_4.setText(_translate("MainWindow", "Search: "))
        self.samplesheet_editor.SSBrowse_Button.setText(_translate("MainWindow", "Browse..."))
        self.samplesheet_editor.SSSave_Button.setText(_translate("MainWindow", "Save and Use"))
        self.pushButton.setText(_translate("MainWindow", "START POWERCALL"))
        self.tabWidget.setTabText(self.tabWidget.indexOf(self.Analisitab), _translate("MainWindow", "Analisi"))
        self.tabWidget.setTabText(self.tabWidget.indexOf(self.panelstab), _translate("MainWindow", "Panel settings"))
        self.tabWidget.setTabText(self.tabWidget.indexOf(self.pipelinestab), _translate("MainWindow", "Pipelines Settings"))
        self.tabWidget.setTabText(self.tabWidget.indexOf(self.toolstab), _translate("MainWindow", "Tools settings"))
        self.menuPowerCall_v2_0.setTitle(_translate("MainWindow", "PowerCall v2.0"))


if __name__ == "__main__":
    app = QtWidgets.QApplication(sys.argv)
    MainWindow = QtWidgets.QMainWindow()
    ui = Ui_MainWindow()
    ui.setupUi(MainWindow)
    #ui.sscomp.initUI()
    MainWindow.show()
    sys.exit(app.exec_())
